src/css.js

```diff
@@ -283,9 +284,13 @@ jQuery.extend({
 				return;
 			}
 
-			// If a number was passed in, add 'px' (except for certain CSS properties)
-			if ( type === "number" && !jQuery.cssNumber[ origName ] ) {
-				value += "px";
+			// If a number was passed in, add the unit (except for certain CSS properties)
+			if ( type === "number" ) {
+				if ( ret && ret[ 3 ] ) {
+					value += ret[ 3 ];
+				} else if ( !jQuery.cssNumber[ origName ] ) {
+					value += "px";
+				}
```

Comment:
You're right, I didn't know what is the best between less charactere or a concat with an empty string `""`.